lib/beaker-puppet/install_utils/puppet5.rb

```diff
@@ -0,0 +1,109 @@
+module BeakerPuppet
+  module Install
+    module Puppet5
+
+      # grab build json from the builds server
+      #
+      # @param [String] project_name Name of the project
+      # @param [String] project_sha SHA ref of the build for the project
+      #
+      # @return [String, String] URL to the build for the SHA of the project,
+      #   build json parsed into a ruby hash
+      def fetch_build_details(project_name, project_sha)
+        buildserver_url = 'http://builds.delivery.puppetlabs.net'
+        buildserver_projectsha_url = "#{buildserver_url}/#{project_name}/#{project_sha}"
+        file_folder = "#{buildserver_projectsha_url}/artifacts"
+        filename = "#{project_sha}.yaml"
+        dst_folder = Dir.pwd
+
+        # fetch & parse file
+        file_local_path = fetch_http_file(file_folder, filename, dst_folder)
+        file_hash = YAML.load_file(file_local_path)
+
+        # getting the relevant data from the file
+        logger.perf_output("YAML HASH BELOW:")
+        logger.perf_output(file_hash)
+        # pp file_hash
+        logger.perf_output("PLATFORM_DATA BELOW:")
+        logger.perf_output(file_hash[:platform_data])
+        return buildserver_projectsha_url, file_hash
+      end
+
+      # get the build artifact URL for a host
+      #
+      # @param [Host] host Host to get artifact URL for
+      # @param [Hash] build_details Details of the build in a hash
+      # @param [String] build_url URL to the build
+      #
+      # @return [String] URL to the build artifact
+      def hosts_artifact_url(host, build_details, build_url)
+        platform_key = host[:platform]
+        variant, version, arch, _ = platform_key.to_array
+        # step: map platform strings into platform_key's for the hash
+        if variant == 'debian' && arch == 'x86_64'
+          platform_key.sub!('x86_64', 'amd64')
+        elsif variant == 'fedora'
+          platform_key.sub!(version, "f#{version}")
+        end
```

Comment:
Sorry to be the downer at the party, but don't we need to fix the naming issue before making these changes? For example, take the fedora case. In beaker+packaging v1, we have an acceptance job with `TEST_TARGET=fedora25-64a`. It's run through `beaker-hostgenerator` to generate a host config like:

```yaml
---
HOSTS:
  fedora25-64-1:
    hypervisor: vmpooler
    platform: fedora-25-x86_64
    template: fedora-25-x86_64
    roles:
    - agent
```

Beaker uses `platform=fedora-25-x86_64` internally. With this PR, we magically change it to `fedora-f25-x86_64`. We then use that key to lookup the build artifact that the packaging repo created.

In the future, packaging v2 will no longer include `f` in the key. Now we need to update beaker to match. But beaker needs to be able to install packages created by v1 and v2 of packaging, e.g. in the puppet-agent LTS pipeline (1.10.x) and the latest 5.x pipeline. Since the beaker change will break older pipelines, we have to take a semver major, or add a feature flag that consumers can opt-into, or add logic to try `f25` and fallback to `25`. In other words, we find ourselves reimplementing the same set of hacks in beaker, that we're trying to avoid in the beaker+packaging epic.

In my mind, beaker should be data-driven. So given a `key`, lookup the packaging artifact/repo in the hash produced by the packaging repo. So long as the key doesn't change for a given OS/arch combo, then we can make changes in the packaging layout without breaking any consumer, e.g. `fc25`. But because this is a bootstrapping problem, we have to get the name right to begin with. I don't think the key necessarily needs to be `platform`, eg it could be something like `host[:packaging_name]`, etc